copy/paste default post breaks roq completely

created default app, runs `quarkus dev` all is great.

duplicates the first .md file in vscode and get a file called: `2024-10-13-the-first-roq copy.md`

and *boom* quarkus starts dumping lots of exceptons. a restart gives you this:

```
2024-12-20 23:41:05,383 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:101)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:121)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:116)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.IllegalArgumentException: Invalid identifier found: [roq-gen/posts/2024-10-13-the-first-roq copy]
        at io.quarkus.qute.EngineImpl.putTemplate(EngineImpl.java:121)
        at io.quarkus.qute.runtime.EngineProducer.processDefaultTemplate(EngineProducer.java:458)
        at io.quarkus.qute.runtime.EngineProducer.<init>(EngineProducer.java:284)
        at io.quarkus.qute.runtime.EngineProducer_Bean.doCreate(Unknown Source)
        at io.quarkus.qute.runtime.EngineProducer_Bean.create(Unknown Source)
        at io.quarkus.qute.runtime.EngineProducer_Bean.create(Unknown Source)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:119)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:38)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:35)
        at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:32)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
        at io.quarkus.arc.impl.ComputingCacheContextInstances.computeIfAbsent(ComputingCacheContextInstances.java:19)
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:35)
        at io.quarkus.qute.runtime.EngineProducer_Bean.get(Unknown Source)
        at io.quarkus.qute.runtime.EngineProducer_Bean.get(Unknown Source)
        at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:559)
        at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:539)
        at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:572)
        at io.quarkus.arc.impl.ArcContainerImpl.instance(ArcContainerImpl.java:339)
        at io.quarkus.qute.runtime.EngineProducer_Observer_Synthetic_ltRbWRttS-zjfhblNzdlh8YOvYg.notify(Unknown Source)
        at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:351)
        at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:333)
        at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:80)
        at io.quarkus.arc.runtime.ArcRecorder.fireLifecycleEvent(ArcRecorder.java:163)
        at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:114)
        at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy_0(Unknown Source)
        at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy(Unknown Source)
        ... 11 more


2024-12-20 23:41:05,399 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure
2024-12-20 23:41:05,399 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure
2024-12-20 23:41:05,441 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: io.quarkus.dev.appstate.ApplicationStartException: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.dev.appstate.ApplicationStateNotification.waitForApplicationStart(ApplicationStateNotification.java:63)
        at io.quarkus.runner.bootstrap.StartupActionImpl.runMainClass(StartupActionImpl.java:142)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:91)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:428)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:55)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:138)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:93)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:101)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:121)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:116)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.IllegalArgumentException: Invalid identifier found: [roq-gen/posts/2024-10-13-the-first-roq copy]
        at io.quarkus.qute.EngineImpl.putTemplate(EngineImpl.java:121)
        at io.quarkus.qute.runtime.EngineProducer.processDefaultTemplate(EngineProducer.java:458)
        at io.quarkus.qute.runtime.EngineProducer.<init>(EngineProducer.java:284)
        at io.quarkus.qute.runtime.EngineProducer_Bean.doCreate(Unknown Source)
        at io.quarkus.qute.runtime.EngineProducer_Bean.create(Unknown Source)
        at io.quarkus.qute.runtime.EngineProducer_Bean.create(Unknown Source)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:119)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:38)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:35)
        at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:32)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
        at io.quarkus.arc.impl.ComputingCacheContextInstances.computeIfAbsent(ComputingCacheContextInstances.java:19)
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:35)
        at io.quarkus.qute.runtime.EngineProducer_Bean.get(Unknown Source)
        at io.quarkus.qute.runtime.EngineProducer_Bean.get(Unknown Source)
        at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:559)
        at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:539)
        at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:572)
        at io.quarkus.arc.impl.ArcContainerImpl.instance(ArcContainerImpl.java:339)
        at io.quarkus.qute.runtime.EngineProducer_Observer_Synthetic_ltRbWRttS-zjfhblNzdlh8YOvYg.notify(Unknown Source)
        at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:351)
        at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:333)
        at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:80)
        at io.quarkus.arc.runtime.ArcRecorder.fireLifecycleEvent(ArcRecorder.java:163)
        at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:114)
        at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy_0(Unknown Source)
        at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy(Unknown Source)
        ... 11 more
2024-12-20 23:41:05,441 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: io.quarkus.dev.appstate.ApplicationStartException: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.dev.appstate.ApplicationStateNotification.waitForApplicationStart(ApplicationStateNotification.java:63)
        at io.quarkus.runner.bootstrap.StartupActionImpl.runMainClass(StartupActionImpl.java:142)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:91)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:428)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:55)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:138)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:93)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:101)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:121)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:116)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.IllegalArgumentException: Invalid identifier found: [roq-gen/posts/2024-10-13-the-first-roq copy]
        at io.quarkus.qute.EngineImpl.putTemplate(EngineImpl.java:121)
        at io.quarkus.qute.runtime.EngineProducer.processDefaultTemplate(EngineProducer.java:458)
        at io.quarkus.qute.runtime.EngineProducer.<init>(EngineProducer.java:284)
        at io.quarkus.qute.runtime.EngineProducer_Bean.doCreate(Unknown Source)
        at io.quarkus.qute.runtime.EngineProducer_Bean.create(Unknown Source)
        at io.quarkus.qute.runtime.EngineProducer_Bean.create(Unknown Source)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:119)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:38)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:35)
        at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:32)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
        at io.quarkus.arc.impl.ComputingCacheContextInstances.computeIfAbsent(ComputingCacheContextInstances.java:19)
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:35)
        at io.quarkus.qute.runtime.EngineProducer_Bean.get(Unknown Source)
        at io.quarkus.qute.runtime.EngineProducer_Bean.get(Unknown Source)
        at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:559)
        at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:539)
        at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:572)
        at io.quarkus.arc.impl.ArcContainerImpl.instance(ArcContainerImpl.java:339)
        at io.quarkus.qute.runtime.EngineProducer_Observer_Synthetic_ltRbWRttS-zjfhblNzdlh8YOvYg.notify(Unknown Source)
        at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:351)
        at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:333)
        at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:80)
        at io.quarkus.arc.runtime.ArcRecorder.fireLifecycleEvent(ArcRecorder.java:163)
        at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:114)
        at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy_0(Unknown Source)
        at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy(Unknown Source)
        ... 11 more


2024-12-20 23:41:05,445 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to recover after failed start: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServerAfterFailedStart(VertxHttpRecorder.java:308)
        at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup.handleFailedInitialStart(VertxHttpHotReplacementSetup.java:73)
        at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.startupFailed(RuntimeUpdatesProcessor.java:1299)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:126)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:428)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:55)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:138)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:93)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:139)
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:78)
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:57)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServerAfterFailedStart(VertxHttpRecorder.java:277)
        ... 9 more
Caused by: java.lang.RuntimeException: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:139)
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleMap(ConfigInstantiator.java:178)
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:121)
        ... 12 more
Caused by: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
        at java.base/java.lang.Class.getConstructor0(Class.java:3761)
        at java.base/java.lang.Class.getConstructor(Class.java:2442)
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:106)
        ... 14 more


2024-12-20 23:41:05,445 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to recover after failed start: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServerAfterFailedStart(VertxHttpRecorder.java:308)
        at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup.handleFailedInitialStart(VertxHttpHotReplacementSetup.java:73)
        at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.startupFailed(RuntimeUpdatesProcessor.java:1299)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:126)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:428)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:55)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:138)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:93)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:139)
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:78)
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:57)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServerAfterFailedStart(VertxHttpRecorder.java:277)
        ... 9 more
Caused by: java.lang.RuntimeException: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:139)
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleMap(ConfigInstantiator.java:178)
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:121)
        ... 12 more
Caused by: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
        at java.base/java.lang.Class.getConstructor0(Class.java:3761)
        at java.base/java.lang.Class.getConstructor(Class.java:2442)
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:106)
        ... 14 more



--
Press [space] to restart, [e] to edit command line args (currently ''), [:] for the terminal, [h] for more options>
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  5.500 s
[INFO] Finished at: 2024-12-20T23:41:06+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.17.5:dev (default-cli) on project roq-with-blog: Failed to run: Dev mode process did not complete successfully -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

expected: Don't fail this hard on totally expected state. Fail soft/be lenient - at least in devmode.

@maxandersen I agree the thrown exception should have more context. What I meant is, this is not Roq printing this:
```
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:139)
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:78)
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:57)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServerAfterFailedStart(VertxHttpRecorder.java:277)
        ... 9 more
Caused by: java.lang.RuntimeException: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:139)
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleMap(ConfigInstantiator.java:178)
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:121)
        ... 12 more
Caused by: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
        at java.base/java.lang.Class.getConstructor0(Class.java:3761)
        at java.base/java.lang.Class.getConstructor(Class.java:2442)
        at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:106)
        ... 14 more

```

--- a/blog/content/docs/basics.adoc
+++ b/blog/content/docs/basics.adoc
@@ -15,11 +15,11 @@ TIP: you may also put it in the Java resources (i.e. `src/main/resources/`).
 
 == Qute and FrontMatter
 
-All templates can use the awesome type-safe https://quarkus.io/guides/qute-reference[Qute] template engine.
+All templates may use the awesome type-safe https://quarkus.io/guides/qute-reference[Qute] template engine.
 
 TIP: Type-safety doesn't make it more complex, it just means that using wrong variables and such will result in build error. This way it won't leak in production.
 
-Templates for layout, documents and pages can declare a FrontMatter (referred as FM data) header delimited by  two `---`.
+Templates for layout, documents and pages may also declare a FrontMatter (referred as FM data) header delimited by  two `---`.
 This header contains yaml data used to configure many things such as:
 
 * Routing
@@ -36,7 +36,7 @@ For your site, you will have one or more kind of pages, this is what we call "la
 * `post`: the layout for blog posts
 * `recipe`: the layout for recipes or whatever
 
-A layout can be included in pages and documents through the `layout` FrontMatter key (e.g. `layout: page`).
+A layout may be specified in pages and documents through the `layout` FrontMatter key (e.g. `layout: page`). By default, `posts` will use the `post` layout and normal pages the `page` layout (this can be configured through link:{site.url('docs/advanced/')}#quarkus-roq-frontmatter_site-page-layout[site configuration]).
 
 TIP: Roq layouts are using the Qute `include` section under the hood to achieve template inheritance. For more details, see the Qute documentation on includes: https://quarkus.io/guides/qute-reference#include_helper[Qute includes]. Unlike partials, layouts can also define Frontmatter data, which is inherited along with the template structure.
 

--- a/docs/modules/ROOT/pages/_includes/quarkus-roq-frontmatter.adoc
+++ b/docs/modules/ROOT/pages/_includes/quarkus-roq-frontmatter.adoc
@@ -77,6 +77,23 @@ endif::add-copy-button-to-env-var[]
 |list of string
 |`**/_**,_**,.**`
 
+a|icon:lock[title=Fixed at build time] [[quarkus-roq-frontmatter_site-page-layout]] [.property-path]##link:#quarkus-roq-frontmatter_site-page-layout[`site.page-layout`]##
+
+[.description]
+--
+The layout to use for normal pages if not specified in FM. When empty, the page will not use a layout when it doesn't specify it in FM. ":theme/" is removed if no theme is defined.
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++SITE_PAGE_LAYOUT+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++SITE_PAGE_LAYOUT+++`
+endif::add-copy-button-to-env-var[]
+--
+|string
+|`:theme/page`
+
 a|icon:lock[title=Fixed at build time] [[quarkus-roq-frontmatter_site-content-dir]] [.property-path]##link:#quarkus-roq-frontmatter_site-content-dir[`site.content-dir`]##
 
 [.description]
@@ -280,6 +297,23 @@ endif::add-copy-button-to-env-var[]
 --
 |boolean
 |`false`
+
+a|icon:lock[title=Fixed at build time] [[quarkus-roq-frontmatter_site-collections-collections-map-layout]] [.property-path]##link:#quarkus-roq-frontmatter_site-collections-collections-map-layout[`site.collections."collections-map".layout`]##
+
+[.description]
+--
+The layout to use if not specified in FM data. When empty, the document will not use a layout when it doesn't specify it in FM. ":theme/" is removed if no theme defined.
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++SITE_COLLECTIONS__COLLECTIONS_MAP__LAYOUT+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++SITE_COLLECTIONS__COLLECTIONS_MAP__LAYOUT+++`
+endif::add-copy-button-to-env-var[]
+--
+|string
+|
 
 |===
 

--- a/docs/modules/ROOT/pages/_includes/quarkus-roq-frontmatter_site.adoc
+++ b/docs/modules/ROOT/pages/_includes/quarkus-roq-frontmatter_site.adoc
@@ -77,6 +77,23 @@ endif::add-copy-button-to-env-var[]
 |list of string
 |`**/_**,_**,.**`
 
+a|icon:lock[title=Fixed at build time] [[quarkus-roq-frontmatter_site-page-layout]] [.property-path]##link:#quarkus-roq-frontmatter_site-page-layout[`site.page-layout`]##
+
+[.description]
+--
+The layout to use for normal pages if not specified in FM. When empty, the page will not use a layout when it doesn't specify it in FM. ":theme/" is removed if no theme is defined.
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++SITE_PAGE_LAYOUT+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++SITE_PAGE_LAYOUT+++`
+endif::add-copy-button-to-env-var[]
+--
+|string
+|`:theme/page`
+
 a|icon:lock[title=Fixed at build time] [[quarkus-roq-frontmatter_site-content-dir]] [.property-path]##link:#quarkus-roq-frontmatter_site-content-dir[`site.content-dir`]##
 
 [.description]
@@ -280,6 +297,23 @@ endif::add-copy-button-to-env-var[]
 --
 |boolean
 |`false`
+
+a|icon:lock[title=Fixed at build time] [[quarkus-roq-frontmatter_site-collections-collections-map-layout]] [.property-path]##link:#quarkus-roq-frontmatter_site-collections-collections-map-layout[`site.collections."collections-map".layout`]##
+
+[.description]
+--
+The layout to use if not specified in FM data. When empty, the document will not use a layout when it doesn't specify it in FM. ":theme/" is removed if no theme defined.
+
+
+ifdef::add-copy-button-to-env-var[]
+Environment variable: env_var_with_copy_button:+++SITE_COLLECTIONS__COLLECTIONS_MAP__LAYOUT+++[]
+endif::add-copy-button-to-env-var[]
+ifndef::add-copy-button-to-env-var[]
+Environment variable: `+++SITE_COLLECTIONS__COLLECTIONS_MAP__LAYOUT+++`
+endif::add-copy-button-to-env-var[]
+--
+|string
+|
 
 |===
 

--- a/roq-frontmatter/deployment/src/main/java/io/quarkiverse/roq/frontmatter/deployment/scan/RoqFrontMatterScanProcessor.java
+++ b/roq-frontmatter/deployment/src/main/java/io/quarkiverse/roq/frontmatter/deployment/scan/RoqFrontMatterScanProcessor.java
@@ -7,6 +7,7 @@ import static io.quarkiverse.roq.util.PathUtils.*;
 import static java.util.function.Predicate.not;
 
 import java.io.IOException;
+import java.io.UncheckedIOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
@@ -26,6 +27,7 @@ import java.util.stream.Stream;
 
 import org.jboss.logging.Logger;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
 
@@ -293,7 +295,7 @@ public class RoqFrontMatterScanProcessor {
                     .filter(RoqFrontMatterScanProcessor::isExtensionSupportedForLayout)
                     .forEach(layoutsConsumer);
         } catch (IOException e) {
-            throw new RuntimeException("Was not possible to scan templates dir %s".formatted(templatesRoot), e);
+            throw new RuntimeException("Error while scanning templates dir %s".formatted(templatesRoot), e);
         }
     }
 
@@ -337,7 +339,7 @@ public class RoqFrontMatterScanProcessor {
                                     .extensionInfo(RoqFrontMatterProcessor.FEATURE)
                                     .build());
                         } catch (IOException e) {
-                            throw new RuntimeException("Error while reading the FrontMatter file %s"
+                            throw new RuntimeException("Error while reading the file %s"
                                     .formatted(p), e);
                         }
                     });
@@ -362,50 +364,65 @@ public class RoqFrontMatterScanProcessor {
                     + resolveOutputExtension(markups, normalizedPath);
             boolean published = type.isPage();
             String id = type.isPage() ? normalizedPath : removeExtension(normalizedPath);
+            final String fullContent;
             try {
-                final String fullContent = Files.readString(file, StandardCharsets.UTF_8);
-                if (hasFrontMatter(fullContent)) {
-                    JsonNode rootNode = mapper.readTree(getFrontMatter(fullContent));
-                    final Map<String, Object> map = mapper.convertValue(rootNode, Map.class);
-                    JsonObject fm = new JsonObject(map);
-
-                    for (RoqFrontMatterDataModificationBuildItem modification : dataModifications) {
-                        fm = modification.modifier().modify(sourcePath, fm);
-                    }
-                    final boolean draft = fm.getBoolean(DRAFT_KEY, false);
-                    if (!config.draft() && draft) {
-                        return;
-                    }
-
-                    final String layoutId = normalizedLayout(config.theme(),
-                            fm.getString(LAYOUT_KEY));
-                    final String content = stripFrontMatter(fullContent);
-                    ZonedDateTime date = parsePublishDate(file, fm, config.dateFormat(), config.timeZone());
-                    final boolean noFuture = !config.future() && (collection == null || !collection.future());
-                    if (date != null && noFuture && date.isAfter(ZonedDateTime.now())) {
-                        return;
-                    }
-                    String dateString = date.format(DateTimeFormatter.ISO_ZONED_DATE_TIME);
-                    PageInfo info = PageInfo.create(id, draft, config.imagesPath(), dateString, content,
-                            sourcePath, quteTemplatePath);
-                    final String layoutTemplate = ROQ_GENERATED_QUTE_PREFIX + layoutId;
-                    final String generatedTemplate = generateTemplate(markups, sourcePath, layoutTemplate, content);
-                    items.add(
-                            new RoqFrontMatterRawTemplateBuildItem(info, layoutId, type, fm, collection, generatedTemplate,
-                                    published));
-                } else {
-                    PageInfo info = PageInfo.create(id, false, config.imagesPath(), null, fullContent, sourcePath,
-                            quteTemplatePath);
-                    items.add(
-                            new RoqFrontMatterRawTemplateBuildItem(info, null, type, new JsonObject(), collection,
-                                    fullContent,
-                                    published));
-                }
+                fullContent = Files.readString(file, StandardCharsets.UTF_8);
             } catch (IOException e) {
-                throw new RuntimeException("Error while reading the FrontMatter file %s"
-                        .formatted(sourcePath), e);
+                throw new UncheckedIOException("Error while reading file: " + sourcePath, e);
             }
+            JsonObject fm;
+            String content = fullContent;
+            if (hasFrontMatter(fullContent)) {
+                try {
+                    fm = readFM(mapper, fullContent);
+                } catch (JsonProcessingException | IllegalArgumentException e) {
+                    throw new RuntimeException("FrontMatter parsing error for file: " + sourcePath, e);
+                }
+                content = stripFrontMatter(fullContent);
+            } else {
+                fm = new JsonObject();
+            }
+            ZonedDateTime date = parsePublishDate(file, fm, config.dateFormat(), config.timeZone());
+            final boolean noFuture = !config.future() && (collection == null || !collection.future());
+            if (date != null && noFuture && date.isAfter(ZonedDateTime.now())) {
+                return;
+            }
+            String dateString = date.format(DateTimeFormatter.ISO_ZONED_DATE_TIME);
+            final String defaultLayout = type.isPage()
+                    ? collection != null ? collection.layout() : config.pageLayout().orElse(null)
+                    : null;
+            final String layoutId = normalizedLayout(config.theme(),
+                    fm.getString(LAYOUT_KEY),
+                    defaultLayout);
+
+            for (RoqFrontMatterDataModificationBuildItem modification : dataModifications) {
+                fm = modification.modifier().modify(sourcePath, fm);
+            }
+            final boolean draft = fm.getBoolean(DRAFT_KEY, false);
+            if (!config.draft() && draft) {
+                return;
+            }
+
+            PageInfo info = PageInfo.create(id, draft, config.imagesPath(), dateString, content,
+                    sourcePath, quteTemplatePath);
+            final String generatedTemplate = generateTemplate(markups, sourcePath, layoutId, content);
+            items.add(
+                    new RoqFrontMatterRawTemplateBuildItem(info, layoutId, type, fm, collection, generatedTemplate,
+                            published));
+
         };
+    }
+
+    @SuppressWarnings("unchecked")
+    private static JsonObject readFM(YAMLMapper mapper, String fullContent)
+            throws JsonProcessingException, IllegalArgumentException {
+        final String frontMatter = getFrontMatter(fullContent);
+        if (frontMatter.isBlank()) {
+            return new JsonObject();
+        }
+        JsonNode rootNode = mapper.readTree(frontMatter);
+        final Map<String, Object> map = mapper.convertValue(rootNode, Map.class);
+        return new JsonObject(map);
     }
 
     private static String normalizePath(String sourcePath) {
@@ -439,25 +456,37 @@ public class RoqFrontMatterScanProcessor {
     private static String generateTemplate(Map<String, QuteMarkupSection> markups, String fileName, String layout,
             String content) {
         StringBuilder template = new StringBuilder();
+
         if (layout != null) {
-            template.append("{#include ").append(layout).append("}\n");
+            template.append("{#include ").append(ROQ_GENERATED_QUTE_PREFIX + layout).append("}\n");
         }
         template.append(find(markups, fileName, Function.identity()).apply(content));
-        template.append("\n{/include}");
+        if (layout != null) {
+            template.append("\n{/include}");
+        }
         return template.toString();
     }
 
-    private static String normalizedLayout(Optional<String> theme, String layout) {
-        if (layout == null) {
-            return null;
-        }
+    private static String normalizedLayout(Optional<String> theme, String layout, String defaultLayout) {
         String normalized = layout;
+        if (normalized == null) {
+            normalized = defaultLayout;
+            if (normalized == null || normalized.isBlank() || "none".equalsIgnoreCase(normalized)) {
+                return null;
+            }
+            normalized = defaultLayout;
+            if (normalized.contains(":theme/") && theme.isEmpty()) {
+                normalized = normalized.replace(":theme/", "");
+            }
+        }
+
         if (normalized.contains(":theme")) {
             if (theme.isPresent()) {
                 normalized = normalized.replace(":theme", theme.get());
             } else {
                 throw new ConfigurationException(
-                        "No theme detected! Using :theme in 'layout: " + layout + " is only possible with a theme installed.");
+                        "No theme detected! Using :theme in 'layout: " + layout
+                                + " is only possible with a theme installed.");
             }
         }
 

--- a/roq-frontmatter/deployment/src/main/java/io/quarkiverse/roq/frontmatter/deployment/scan/RoqFrontMatterScanProcessor.java
+++ b/roq-frontmatter/deployment/src/main/java/io/quarkiverse/roq/frontmatter/deployment/scan/RoqFrontMatterScanProcessor.java
@@ -60,6 +60,7 @@ public class RoqFrontMatterScanProcessor {
     public static final String LAYOUTS_DIR = "layouts";
     public static final String THEME_LAYOUTS_DIR_PREFIX = "theme-";
     public static final String TEMPLATES_DIR = "templates";
+    public static final Pattern NON_PATH_CHAR_PATTERN = Pattern.compile("[^a-zA-Z0-9_\\\\/.\\-]");
 
     @BuildStep
     void scan(RoqProjectBuildItem roqProject,
@@ -356,10 +357,11 @@ public class RoqFrontMatterScanProcessor {
             TemplateType type) {
         return file -> {
             String sourcePath = toUnixPath(root.relativize(file).toString());
-            String quteTemplatePath = ROQ_GENERATED_QUTE_PREFIX + removeExtension(sourcePath)
-                    + resolveOutputExtension(markups, sourcePath);
+            String normalizedPath = normalizePath(sourcePath);
+            String quteTemplatePath = ROQ_GENERATED_QUTE_PREFIX + removeExtension(normalizedPath)
+                    + resolveOutputExtension(markups, normalizedPath);
             boolean published = type.isPage();
-            String id = type.isPage() ? sourcePath : removeExtension(sourcePath);
+            String id = type.isPage() ? normalizedPath : removeExtension(normalizedPath);
             try {
                 final String fullContent = Files.readString(file, StandardCharsets.UTF_8);
                 if (hasFrontMatter(fullContent)) {
@@ -404,6 +406,10 @@ public class RoqFrontMatterScanProcessor {
                         .formatted(sourcePath), e);
             }
         };
+    }
+
+    private static String normalizePath(String sourcePath) {
+        return NON_PATH_CHAR_PATTERN.matcher(sourcePath).replaceAll("-");
     }
 
     protected static ZonedDateTime parsePublishDate(Path file, JsonObject frontMatter, String dateFormat,

--- a/roq-frontmatter/runtime/src/main/java/io/quarkiverse/roq/frontmatter/runtime/config/ConfiguredCollection.java
+++ b/roq-frontmatter/runtime/src/main/java/io/quarkiverse/roq/frontmatter/runtime/config/ConfiguredCollection.java
@@ -1,4 +1,4 @@
 package io.quarkiverse.roq.frontmatter.runtime.config;
 
-public record ConfiguredCollection(String id, boolean hidden, boolean future) {
+public record ConfiguredCollection(String id, boolean hidden, boolean future, String layout) {
 }

--- a/roq-frontmatter/runtime/src/main/java/io/quarkiverse/roq/frontmatter/runtime/config/RoqSiteConfig.java
+++ b/roq-frontmatter/runtime/src/main/java/io/quarkiverse/roq/frontmatter/runtime/config/RoqSiteConfig.java
@@ -21,7 +21,7 @@ public interface RoqSiteConfig {
     String CONTENT_DIR = "content";
     String STATIC_DIR = "static";
     String IGNORED_FILES = "**/_**,_**,.**";
-    List<ConfiguredCollection> DEFAULT_COLLECTIONS = List.of(new ConfiguredCollection("posts", false, false));
+    List<ConfiguredCollection> DEFAULT_COLLECTIONS = List.of(new ConfiguredCollection("posts", false, false, ":theme/post"));
 
     /**
      * The root path of your site (e.g. /blog) relative the quarkus http root path.
@@ -53,6 +53,15 @@ public interface RoqSiteConfig {
      */
     @WithDefault(IGNORED_FILES)
     List<String> ignoredFiles();
+
+    /**
+     * The layout to use for normal pages if not specified in FM.
+     * When empty, the page will not use a layout when it doesn't specify it in FM.
+     *
+     * ":theme/" is removed if no theme is defined.
+     */
+    @WithDefault(":theme/page")
+    Optional<String> pageLayout();
 
     /**
      * The directory which contains content (pages and collections) in the Roq site directory.
@@ -122,7 +131,9 @@ public interface RoqSiteConfig {
             return DEFAULT_COLLECTIONS;
         }
         return collectionsMap().entrySet().stream().filter(e -> e.getValue().enabled())
-                .map(e -> new ConfiguredCollection(e.getKey(), e.getValue().hidden(), e.getValue().future())).toList();
+                .map(e -> new ConfiguredCollection(e.getKey(), e.getValue().hidden(), e.getValue().future(),
+                        e.getValue().layout().orElse(null)))
+                .toList();
     }
 
     interface CollectionConfig {
@@ -144,5 +155,13 @@ public interface RoqSiteConfig {
          */
         @WithDefault("false")
         boolean hidden();
+
+        /**
+         * The layout to use if not specified in FM data.
+         * When empty, the document will not use a layout when it doesn't specify it in FM.
+         *
+         * ":theme/" is removed if no theme defined.
+         */
+        Optional<String> layout();
     }
 }

--- a/roq-frontmatter/runtime/src/main/java/io/quarkiverse/roq/frontmatter/runtime/model/Page.java
+++ b/roq-frontmatter/runtime/src/main/java/io/quarkiverse/roq/frontmatter/runtime/model/Page.java
@@ -71,7 +71,7 @@ public interface Page {
      * The page title (from FM data)
      */
     default String title() {
-        return data().getString(FM_TITLE);
+        return data().getString(FM_TITLE, sourcePath());
     }
 
     /**

